src/server/req_manager.c

```diff
@@ -1724,7 +1725,7 @@ mgr_server_unset(struct batch_request *preq)
 			else if (strcasecmp(plist->al_name, ATTR_rescdflt) == 0) {
 				if (plist->al_resc != NULL && strcasecmp(plist->al_resc, "ncpus") == 0) {
 					svrattrl *tm_list;
-					tm_list = attrlist_create(plist->al_name, "ncpus", 1);
+					tm_list = attrlist_create(plist->al_name, "ncpus", 2);
```

Comment:
Why are you changing the resources_default.ncpus default from 1 to 2?  Shouldn't it be 1?

Reply:
Well, 2 is not the resource value it is the amount of memory needed to store the value(which in our case is "1"). I'm changing it to 2 because it will need space for a string termination character as well.

Reply:
The svrattrl structure includes al_valln to keep track of the length of the value, which may contain multiple NULL bytes. Anything referencing svrattrl.al_value _should_ ensure it does not run past the end of the buffer. That said, it would be easy to mistakenly overrun the buffer if it is not NULL terminated. If we want to address this risk for all cases, we should probably do so in attrlist_create(). Just take whatever value size is passed and increment by one for the NULL terminator.

In the case of unsetting the default ncpus, we know we only need room for "1\0", but what if the value is changed again later? Do we create another attribute, or do we reuse the one created here? If we reuse this one, shouldn't there be enough space allocated to hold larger numbers?

I'll also note that strcpy() is used in attrlist_create() rather than rather than snprintf().

Reply:
@mike0042 I like your suggestion of incrementing the length in attrlist_create function. 
In case of unsetting ncpus, if in future we plan to reset it back to a bigger numerical value, we will have to allocate enough space for it. Although, it seems unlikely that we will change the resources_default.ncpus to be a large value.
This attribute list is created only temporarily to make sure the mgr_set_attr makes a copy of it and it is immediately deleted afterward. I don't think we will have to reuse the attribute, it will always be created afresh.
For this code change, I'll increase the value in any case.